docs/topic-guides/training-units.txt

```diff
@@ -0,0 +1,180 @@
+trains a single trial on 900 batches, where a batch is defined by your model's dataloader.
+
+.. code:: yaml
+
+  hyperparameters:
+    global_batch_size: 64
+  searcher:
+    name: single
+    metric: validation_error
+    max_length:
+      batches: 900
+    smaller_is_better: true
+
+To direct Determined to validate at least every 100 batches and checkpoint at least every
+50 batches, the snippet below also sets ``min_validation_period`` and ``min_checkpoint_period``.
+
+.. code:: yaml
+
+  hyperparameters:
+    global_batch_size: 64
+  min_validation_period:
+    batches: 100
+  min_checkpoint_period:
+    batches: 50
+  searcher:
+    name: single
+    metric: validation_error
+    max_length:
+      batches: 900
+    smaller_is_better: true
+
+*********
+ Records
+*********
+
+If you are more familiar expressing training lengths in terms of records (or synonymous units,
+such as inputs or samples), in Determined you can specify the
+``max_length`` in terms of records. The snippet below describes a searcher
+that trains a single trial on 57,600 records.
+
+.. code:: yaml
+
+  hyperparameters:
+    global_batch_size: 64
+  searcher:
+    name: single
+    metric: validation_error
+    max_length:
+      records: 57600
+    smaller_is_better: true
+
+To direct Determined to validate at least every 6400 records and checkpoint at least every
+3200 records, the snippet below also sets ``min_validation_period`` and ``min_checkpoint_period``.
+
+.. code:: yaml
+
+  hyperparameters:
+    global_batch_size: 64
+  min_validation_period:
+    records: 6400
+  min_checkpoint_period:
+    records: 3200
+  searcher:
+    name: single
+    metric: validation_error
+    max_length:
+      records: 57600
+    smaller_is_better: true
+
+********
+ Epochs
+********
+
+Finally, if you prefer expressing training lengths in terms of epochs, in Determined you can specify
+the ``max_length`` in terms of epochs. In the case of epochs,
+:ref:`records_per_epoch <config-records-per-epoch>` must also be specified. The snippet below
+describes a searcher that trains a single trial on a single epoch.
+
+.. code:: yaml
+
+  records_per_epoch: 57600
+  hyperparameters:
+    global_batch_size: 64
+  searcher:
+    name: single
+    metric: validation_error
+    max_length:
+      epochs: 1
+    smaller_is_better: true
+
+To direct Determined to validate at least every 6400 records and checkpoint at least every
+3200 records, the snippet below also sets ``min_validation_period`` and ``min_checkpoint_period``.
+
+.. code:: yaml
+
+  hyperparameters:
+    global_batch_size: 64
+  min_validation_period:
+    records: 6400
+  min_checkpoint_period:
+    records: 3200
+  searcher:
+    name: single
+    metric: validation_error
+    max_length:
+      records: 57600
+    smaller_is_better: true
+
+.. note::
+
+  Determined will truncate any partial batches requested as a result of a configuration in records
```

Comment:
I'm not sure "truncate" is the weird we want -- we drop the entire partial batch. So I'd use "drop", "ignore", or "omit".